src/controllers/SitemapController.php

```diff
@@ -84,4 +91,41 @@ private function buildSitemapfile($sitemapFile)
         // write sitemap files
         $sitemap->write();
     }
+    
+    /**
+     * Get full relative URI by NavItem
+     *
+     * @author Robert Kuznetsov <robert@malanka.pro>
+     *
+     * @param NavItem $navItem object
+     *
+     * @return return string
+     */
+    private function getRelativeUriByNavItem($navItem)
+    {
+        $fullUriPath = $navItem->alias;
+        $language = $navItem->lang->short_code;
+        $parentNavId = $navItem->nav->attributes['parent_nav_id'];
+        $nestingIndex = 0;
+        while ($parentNavId) {
+            $parentNav = Nav::find()->where([
+                'is_deleted' => false,
+                'is_offline' => false,
+                'is_draft' => false,
+                'id' => $parentNavId,
+            ])->one();
+
+            $parentNavItem = $parentNav->getActiveLanguageItem()->one();
+            $alias = $parentNavItem->attributes['alias'];
+            $fullUriPath = $alias . '/' . $fullUriPath;
+            $parentNavId = $parentNav->attributes['parent_nav_id'];
+            $nestingIndex += 1;
+            if ($nestingIndex >= $this->maxPathNesting) {
+                // TODO create Exception or other action for notify to customers
```

Comment:
why did you add this limit? the nesting is limited by data in CMS, I do not think adding a limit with arbitrary value here makes sense.

Reply:
I did not know CMS was somehow limited to nesting for URL path.

I did not find specific recommendations on the level of nesting but I have a few SEO experts said that it is better to make nesting no more than 10.

Reply:
sure, that is about SEO and best practise, but there is no point in limiting it in the sitemap.xml.